src/cli/add.rs

```diff
@@ -79,14 +77,34 @@ impl SpecType {
 pub async fn execute(args: Args) -> miette::Result<()> {
     let mut project = Project::load_or_else_discover(args.manifest_path.as_deref())?;
     let spec_type = SpecType::from_args(&args);
-    add_specs_to_project(&mut project, args.specs, spec_type, args.no_install).await
+    let spec_platforms = args.platforms;
+
+    if let Some(platforms) = &spec_platforms {
+        // Add the platform if it is not already present
+        let platforms_to_add = platforms
+            .iter()
+            .filter(|p| !project.platforms().contains(p))
+            .cloned()
+            .collect::<Vec<Platform>>();
+        project.add_platforms(platforms_to_add.iter())?;
+    }
+
+    add_specs_to_project(
+        &mut project,
+        args.specs,
+        spec_type,
+        args.no_install,
+        spec_platforms,
+    )
+    .await
 }
 
 pub async fn add_specs_to_project(
     project: &mut Project,
     specs: Vec<MatchSpec>,
     spec_type: SpecType,
     no_install: bool,
+    specs_platforms: Option<Vec<Platform>>,
```

Comment:
I would expect this to be the argument type:

```suggestion
    specs_platforms: Vec<Platform>,
```

Reply:
Yep changed it so that it works differently